source/Cosmos.System2/Audio/AudioMixer.cs

```diff
@@ -0,0 +1,177 @@
+﻿using Cosmos.HAL.Audio;
+using Cosmos.System.Audio.IO;
+using System;
+using System.Collections.Generic;
+
+namespace Cosmos.System.Audio {
+    /// <summary>
+    /// An audio mixer is responsible for mixing several streams of audio together.
+    /// </summary>
+    /// <remarks>
+    /// The <see cref="AudioMixer"/> class does not implement any kind of compressor or
+    /// limiter, and as such, the resulting audio may introduce hard-clipping.
+    /// </remarks>
+    public class AudioMixer : AudioStream {
+        uint pos = 0;
+        uint sampleRate;
+        AudioBuffer mixBuffer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AudioMixer"/> class,
+        /// with the specified initial internal mixing buffer size and sample format.
+        /// </summary>
+        /// <remarks>
+        /// This constructor should be used when the sizes and formats of the
+        /// buffers that will be given to the mixer are known. If there is a
+        /// mis-match between the mixing format and/or the buffer size and the given buffer,
+        /// the internal mixing buffer will be re-initialized with the right values.
```

Comment:
Why not throw an exception in this case?

Reply:
Honestly, this whole constructor can be deleted, it was made before the `AudioMixer` class supported writing to buffers of different sizes :P